Mathlib/Tactic/Positivity/Basic.lean

```diff
@@ -328,28 +330,31 @@ Since the output type of `Int.natAbs` is `ℕ`, the nonnegative case is handled
 `positivity` tactic.
 -/
 @[positivity Int.natAbs _]
-def evalNatAbs : PositivityExt where eval {_u _α} _zα _pα (e : Q(ℕ)) := do
-  let ~q(Int.natAbs $a) := e | throwError "not Int.natAbs"
-  let zα' : Q(Zero Int) := q(inferInstance)
-  let pα' : Q(PartialOrder Int) := q(inferInstance)
-  let ra ← core zα' pα' a
-  match ra with
-  | .positive pa =>
-    pure (.positive (q(int_natAbs_pos $pa) : Expr))
-  | .nonzero pa =>
-    pure (.positive (q(Int.natAbs_pos.mpr $pa) : Expr))
-  | .nonnegative _pa =>
-    pure .none
-  | .none =>
-    pure .none
+def evalNatAbs : PositivityExt where eval {u α} _zα _pα e := do
+  match u, α, e with
+  | 0, ~q(ℕ), ~q(Int.natAbs $a) =>
+    let zα' : Q(Zero Int) := q(inferInstance)
+    let pα' : Q(PartialOrder Int) := q(inferInstance)
+    let ra ← core zα' pα' a
+    match ra with
+    | .positive pa =>
+      assertInstancesCommute
+      pure (.positive q(int_natAbs_pos $pa))
+    | .nonzero pa =>
+      assertInstancesCommute
+      pure (.positive q(Int.natAbs_pos.mpr $pa))
+    | .nonnegative _pa =>
+      pure .none
+    | .none =>
+      pure .none
```

Comment:
```suggestion
    | _ => pure .none
```

Reply:
I'd prefer to keep this change focused on the match changes and nothing more (these lines vanish from the diff once you ignore whitespace).